app/models/code_owner_project.rb

```diff
@@ -0,0 +1,25 @@
+# == Schema Information
+#
+# Table name: code_owner_projects
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  project_id :bigint           not null
+#  user_id    :bigint           not null
+#
+# Indexes
+#
+#  index_code_owner_projects_on_project_id  (project_id)
+#  index_code_owner_projects_on_user_id     (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (project_id => projects.id)
+#  fk_rails_...  (user_id => users.id)
+#
+
+class CodeOwnerProject < ApplicationRecord
```

Comment:
Absolutely not against this solution, but have you considered using the `UsersProject` entity for this? Maybe just adding a boolean column `code_owner` could have worked.
Again, just an observation that may have come too late, not that we should go for it now.